borsh-schema-derive-internal/src/enum_schema.rs

```diff
@@ -45,6 +46,20 @@ pub fn process_enum(input: &ItemEnum, cratename: Ident) -> syn::Result<TokenStre
                     #ident ,
                 }
             });
+
+        match &mut anonymous_struct.fields {
+            Fields::Named(named) => {
+                for field in &mut named.named {
+                    field.attrs = filter_skip(&field.attrs);
```

Comment:
nit: There is `fields: variant.fields.clone(),` above, so you can use either of the three paths:

1. Instead of cloning `variant.fields` on construction, filter it out before `anonymous_struct` construction
2. Pass owned `field.attrs` instead of a reference, and use [`Vec::retain`](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.retain)
3. Pass `&mut field.attrs`, don't return anything, and use [`Vec::retain`](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.retain)